### PR TITLE
LED Support for DellEMC Z9100

### PR DESCRIPTION
**Pull request by paavaanan**
**- What I did**
Added support for Switchport LED for DellEMC Z9100 switches

**- How I did it**
Added led_proc_init.soc file for T0 and T1 Topologies. 

**- How to verify it**

1. Check Loopback/OIR/ifconifg Ethernet up/down/ tests for all ports. 
2. 50G will be indicated as amber.
3. 100G wille be indicated as green.

### CODE REVIEW

File: platform/broadcom/sonic-platform-modules-dell/z9100/scripts/z9100_platform.sh [hunk 1]
```diff
@@ -151,6 +151,27 @@ reset_muxes() {
...
+    platform=$(/usr/local/bin/sonic-cfggen -H -v DEVICE_METADATA.localhost.platform)
+    hwsku=$(cat /etc/sonic/config_db.json | grep -A2 "DEVICE_METADATA" | grep "hwsku" | cut -d ":" -f2 | sed 's/"//g' | sed 's/,//g'| xargs )
+
+    if [ -z "$hwsku" ]; then
+          #Check minigraph for hwsku
+          cat /etc/sonic/minigraph.xml | grep $T1  > /dev/null
+          if [ $? -eq 0 ]; then
+                  hwsku=$T1
+          else
+                  hwsku=$T2
```
Comment (prsunny) [inline platform/broadcom/sonic-platform-modules-dell/z9100/scripts/z9100_platform.sh]: Is it `T0` here?

File: platform/broadcom/sonic-platform-modules-dell/z9100/scripts/z9100_platform.sh [hunk 2]
```diff
@@ -171,6 +192,13 @@ if [[ "$1" == "init" ]]; then
     echo $value > /sys/class/i2c-adapter/i2c-15/15-003e/qsfp_lpmode
     echo $value > /sys/class/i2c-adapter/i2c-16/16-003e/qsfp_lpmode
 
+
+    #Copy led_proc_init.soc
+    switch_port_led
+    if [ -e $led_proc_init ] && [ ! -e $device/$platform/led_proc_init.soc ]; then
+      cp $led_proc_init $device/$platform
```
Comment (jleveque) [inline platform/broadcom/sonic-platform-modules-dell/z9100/scripts/z9100_platform.sh]: I'm not a fan of copying the file from the HwSKU directory into to the platform directory. If the LED functionality need to differ based on SKU, then we may want to consider moving all led_proc_init.soc files into the respective HwSKU directories. For SKUs which share a common led_proc_init.soc file, we can symlink them.

I'm not sure that this should hold up this PR, but it's something I wanted to at least note here for posterity.

File: platform/broadcom/sonic-platform-modules-dell/z9100/scripts/z9100_platform.sh (hunk 1 above)
Comment (paavaanan) [reply platform/broadcom/sonic-platform-modules-dell/z9100/scripts/z9100_platform.sh]: Addressed in new commint @prsunny

File: platform/broadcom/sonic-platform-modules-dell/z9100/scripts/z9100_platform.sh (hunk 2 above)
Comment (paavaanan) [reply platform/broadcom/sonic-platform-modules-dell/z9100/scripts/z9100_platform.sh]: Joe, thanks for the view. I changed copy to symlink in new commit. Right now as per the design ledinit "platform/broadcom/docker-syncd-brcm/supervisord.conf" picks led_proc_init.soc in platform specific locaation than HWSKU

**command=/usr/bin/bcmcmd -t 60 "rcload /usr/share/sonic/platform/led_proc_init.soc"**

Comment (jleveque) [reply platform/broadcom/sonic-platform-modules-dell/z9100/scripts/z9100_platform.sh]: Yes, I understand that the current implementation looks for the led_proc_init.soc file in the platform directory. What I was noting for the future is that maybe we should consider moving the files into HwSKU directories and symlinking them if they share the same file.

I wasn't asking you to replace your copy operation with symlinking, but I'm OK with the change. My comment was about refactoring the mechanism completely in the future.

Comment (paavaanan) [reply platform/broadcom/sonic-platform-modules-dell/z9100/scripts/z9100_platform.sh]: Thanks Joe. Got it!

**Comment by jleveque** [comment]
@paavaanan: WIll you also need to make a similar PR on the master branch (which should also be able to be cherry-picked into the 201811 branch?

**Comment by paavaanan** [comment]
Joe, z9100_platform.sh do have some differences. Will raise a seperate PR to Master Branch. Better cherry-pick from that PR for 201811 branch

**Comment by jleveque** [comment]
Closing this PR as the 201803 branch does not officially support Z9100. Similar PR https://github.com/Azure/sonic-buildimage/pull/2799 was merged into master and will be cherry-picked into the 201811 branch.

**Pull request closed without merging**